### PR TITLE
refactor: upgrade to release-please v13

### PR DESCRIPTION
**Pull request by chingor13**
This updates the release-please bot to use v13 of release-please. We now expose more options like `versioning`, `changelogType`.

We also allow running on the all matching configurations for a branch so you can have more than one configuration per branch.

### CODE REVIEW

**Comment by chingor13** [comment]
@bcoe Can probably still review, but we'll wait until the new year to merge when we'll for sure be around to debug.

**Comment by bcoe** [comment]
@chingor13 probably won't be able to look until new year, I'm glad we have this queued up.

File: packages/release-please/test/release-please.ts
Comment (bcoe) [inline packages/release-please/test/release-please.ts:762]: Do we have anythiung that confirms we default to the appropriate branch if not config provided? Is there a test for this in release-please itself?

**Review by bcoe — CHANGES_REQUESTED** [review]
I'm a little worried about dropping the tests for choosing an appropriate branch by default, and allowing this to be overridden in config, is this behavior still caught anywhere.

**Review by bcoe — APPROVED** [review]

**Pull request merged**
